<?xml version="1.0" encoding="utf-8"?>
<launch>
    <arg name="veh" default="$(env VEHICLE_NAME)" />
    <arg name="config" default="default" />
    <arg name="param_file_name" default="default" />
    
    <group ns="$(arg veh)">
        <node name="enhanced_object_detection_node" 
              pkg="object_detection" 
              type="enhanced_object_detection_node.py" 
              output="screen" 
              clear_params="true">
            
            <rosparam file="$(find object_detection)/config/enhanced_object_detection_node/$(arg param_file_name).yaml" command="load"/>
        </node>
    </group>
</launch>